try_install_modified.py

```diff
@@ -0,0 +1,165 @@
+"""Attempt to update datasets in all database management systems.
+This module, when run, attempts to install datasets from modified Retriever
+scripts in the /scripts folder (except for those listed in IGNORE), for each
+engine in ENGINE_LIST() from __init__.py. In other words, it trys to install
+using all possible combinations of database platform and script and checks
+to see if there are any errors. It does not check the values in the database.
+"""
+from __future__ import print_function
+from __future__ import absolute_import
+from future import standard_library
+
+standard_library.install_aliases()
+import os
+import sys
+import urllib.request
+import urllib.parse
+import urllib.error
+from distutils.version import LooseVersion
+from imp import reload
+from retriever.lib.tools import choose_engine
+from retriever import MODULE_LIST, ENGINE_LIST, SCRIPT_LIST
+from version import get_module_version
+
+reload(sys)
+if hasattr(sys, 'setdefaultencoding'):
+    sys.setdefaultencoding('latin-1')
+if os.name == "nt":
+    os_password = "Password12!"
+else:
+    os_password = ""
+
+MODULE_LIST = MODULE_LIST()
+ENGINE_LIST = ENGINE_LIST()
+SCRIPT_LIST = SCRIPT_LIST()
+TEST_ENGINES = {}
+IGNORE = [
+    "forest-inventory-analysis",
+    "bioclim",
+    "prism-climate",
+    "vertnet",
+    "NPN",
+    "mammal-super-tree"
+]
+IGNORE = [dataset.lower() for dataset in IGNORE]
+UPSTREAM_VERSIONS = {}
+
+
+def to_string(value_to_str):
+    if sys.version_info >= (3, 0, 0):
+        return value_to_str.decode("UTF-8")
+    else:
+        return value_to_str
+
+
+def get_modified_scripts():
+    """Get modified script list, using version.txt in repo and master upstream"""
+    modified_list = []
+    version_file = urllib.request.urlopen("https://raw.githubusercontent.com/weecology/retriever/master/version.txt")
+    local_repo_scripts = get_module_version()  # local repo versions
+
+    version_file.readline()
+    for line in version_file.readlines():
+        master_script_name, master_script_version = to_string(line).lower().strip().split(",")
+        UPSTREAM_VERSIONS[master_script_name] = master_script_version
+
+    for item in local_repo_scripts:
+        local_script, local_version = item.lower().split(",")
+        # check for new scripts or a change in versions for present scripts
+        # repo script versions compared with upstream.
+        if local_script not in UPSTREAM_VERSIONS.keys():
+            modified_list.append( os.path.basename(local_script).split('.')[0])
+        else:
+            if LooseVersion(local_version) == UPSTREAM_VERSIONS[local_script]:
+                pass
+            else:
+                modified_list.append( os.path.basename(local_script).split('.')[0])
```

Comment:
What about shortening Lines 72-76 to:

```
elif LooseVersion(local_version) != UPSTREAM_VERSIONS[local_script]:
    modified_list.append( os.path.basename(local_script).split('.')[0])
```

This is shorter, removes a layer of nesting, and makes is clearer that we're doing two checks "Is it new" and "Has it changed" and adding it to `modified_list` if either of those is true.